server/src/test/java/org/opensearch/index/translog/TranslogDeletionPolicyTests.java

```diff
@@ -128,6 +163,34 @@ public void testTotalFilesRetention() throws Exception {
         }
     }
 
+    public void testBySizeAndRetentionLease() throws Exception {
+        long now = System.currentTimeMillis();
+        Tuple<List<TranslogReader>, TranslogWriter> readersAndWriter = createReadersAndWriter(now);
+        List<BaseTranslogReader> allGens = new ArrayList<>(readersAndWriter.v1());
+        allGens.add(readersAndWriter.v2());
+        try {
+            int selectedReader = randomIntBetween(0, allGens.size() - 1);
+            final long selectedGeneration = allGens.get(selectedReader).generation;
+
+            // Retaining seqno is part of lower gen
+            long size = allGens.stream().skip(selectedReader).map(BaseTranslogReader::sizeInBytes).reduce(Long::sum).get();
+            Supplier<RetentionLeases> retentionLeasesSupplier = createRetentionLeases(now, 0L, (selectedGeneration - 1) * TOTAL_OPS_IN_GEN);
+            TranslogDeletionPolicy deletionPolicy = new MockDeletionPolicy(now, size, Long.MIN_VALUE,
+                Integer.MIN_VALUE, retentionLeasesSupplier);
+            assertThat(deletionPolicy.minTranslogGenRequired(readersAndWriter.v1(), readersAndWriter.v2()), equalTo(selectedGeneration));
+
+            // Retention lease is part of higher gen
+            retentionLeasesSupplier = createRetentionLeases(now, selectedGeneration * TOTAL_OPS_IN_GEN, allGens.size() * TOTAL_OPS_IN_GEN - 1);
+            deletionPolicy = new MockDeletionPolicy(now, size, Long.MIN_VALUE,
+                Integer.MIN_VALUE, retentionLeasesSupplier);
+            assertThat(deletionPolicy.minTranslogGenRequired(readersAndWriter.v1(), readersAndWriter.v2()), equalTo(selectedGeneration));
```

Comment:
Would be good to assert that all other policies that like age, size etc. require a higher gen, but retention lease requires a lower one.

Reply:
Added asserts for the rest of the policies in the same test.